### PR TITLE
fix(docs): security practices guide updated

### LINKED ISSUE #52: adding firewall config description in validators guide

### PR DESCRIPTION
**Pull request by PastaHash**
Closes: #52 

## Description

- firewall config descriptions changed

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: docs/guides/validators/security_best_practices.md [hunk 1]
```diff
@@ -373,25 +378,30 @@ You can find used ports in your node config files and make your own firewall rul
...
-- EMV WebSocket `0.0.0.0:8546`
-- Tendermint RPC `tcp://localhost:26657`
-- ABCI application `tcp://127.0.0.1:26658`
-- pprof (Go pkg) `localhost:6060`
-- p2p incoming connections `tcp://localhost:26656`
-- Prometheus `:26660`
+| port | service | used by | rpc node | validator |
+|---|---|---|---|---|
+| 1317 | API server |  | ✅ | ❌ 
+| 8080 | Rosseta API |  | ❌ | ❌ |
```
Comment (onsails) [inline docs/guides/validators/security_best_practices.md]: why not allow rosetta for rpc node?

File: docs/guides/validators/security_best_practices.md [hunk 2]
```diff
@@ -373,25 +378,30 @@ You can find used ports in your node config files and make your own firewall rul
...
+| 26656 | p2p incoming connections |  | ✅ | ✅ |
+| 26660 | Prometheus |  | ❌ | ❌ |
 
-::: tip
+<br>
+
+:::danger
 
-For security reasons, we recommend that you use only the services and ports necessary for your node to work.
+* Port `26657` is used for working with node CLI. For security reasons we recommended to close this port access from public network.
```
Comment (onsails) [inline docs/guides/validators/security_best_practices.md]: it's a bit confusing to note this only for tendermint rpc but not for other RPCs.

File: docs/guides/validators/security_best_practices.md [hunk 3]
```diff
@@ -373,25 +378,30 @@ You can find used ports in your node config files and make your own firewall rul
...
-- Prometheus `:26660`
+| port | service | used by | rpc node | validator |
+|---|---|---|---|---|
+| 1317 | API server |  | ✅ | ❌ 
+| 8080 | Rosseta API |  | ❌ | ❌ |
+| 9090 | gRPC server |  | ✅ | ❌ |
+| 1317 | gRPC web |  | ✅ | ❌ |
+| 8545 | EMV RPC HTTP |  | ✅ | ❌ |
+| 8546 | EMV RPC WebSocket |  | ✅ | ❌ |
+| **26657** * | Tendermint RPC |  | ✅ | ❌ |
```
Comment (onsails) [inline docs/guides/validators/security_best_practices.md]: why we are BOLDing this one?
I believe there is no "should" when in comes to ports on validators. Only "must". So having this one bold is a bit confusing 'cause it makes sense that others are just **should** be closed.

**Review by onsails — CHANGES_REQUESTED** [review]

File: docs/guides/validators/security_best_practices.md (hunk 1 above)
Comment (PastaHash) [reply docs/guides/validators/security_best_practices.md]: changed to allowed

File: docs/guides/validators/security_best_practices.md (hunk 2 above)
Comment (PastaHash) [reply docs/guides/validators/security_best_practices.md]: Changed with text about all listed ports in table

File: docs/guides/validators/security_best_practices.md (hunk 3 above)
Comment (PastaHash) [reply docs/guides/validators/security_best_practices.md]: bolding removed

File: docs/guides/validators/security_best_practices.md [hunk 4]
```diff
@@ -447,9 +460,11 @@ sudo iptables -A INPUT -p tcp --dport 26657 -j DROP
 sudo iptables -A INPUT -p tcp --dport 26658 -j DROP
 ```
 
-`Port` 26657 is using by default for RPC server and if you want you can limit incoming connections too.
+::: tip
+
+If you want to use haqq node CLI remotely you might change IP address `laddr` to 0.0.0.0
```
Comment (onsails) [inline docs/guides/validators/security_best_practices.md]: Let's stick to just minimum and necessary.
We should give an instruction on how to secure validator on firewall level. This one is confusing and can cause operators to expose something which should not exposed.

File: docs/guides/validators/security_best_practices.md [hunk 5]
```diff
@@ -473,6 +488,17 @@ We don't recommend limiting connections to this `port` (26656) because it is use
 laddr = "tcp://0.0.0.0:26656"
 ```
 
+Removing all other traffic 
+It is extremely important to use the DROP target for all other traffic **after** defining the --dport rules. This will prevent unauthorized access to the server through other open ports. To do this, simply type:
```
Comment (onsails) [inline docs/guides/validators/security_best_practices.md]: Instead of explaining this in plain english, just provide a bash instruction set with correct order like it's done for ufw.

File: docs/guides/validators/security_best_practices.md [hunk 6]
```diff
@@ -332,18 +332,23 @@ sudo invoke-rc.d iptables-persistent save
 
 ### Rule for ssh connection
 
-:::danger
+::: tip
 
-This is just an **example** 
+We want to keep our SSH port open from the 192.168.1.3 network we blocked in the above case. That is we only want to allow those packets coming from 192.168.1.3 and which wants to go to the port 22.
```
Comment (onsails) [inline docs/guides/validators/security_best_practices.md]: It's confusing. We didn't block anything yet. There is just fail2ban which is not block per se. It bans connections only after auth failure.

**Review by onsails — CHANGES_REQUESTED** [review]

File: docs/guides/validators/security_best_practices.md (hunk 4 above)
Comment (PastaHash) [reply docs/guides/validators/security_best_practices.md]: replaced

File: docs/guides/validators/security_best_practices.md (hunk 5 above)
Comment (PastaHash) [reply docs/guides/validators/security_best_practices.md]: changed

File: docs/guides/validators/security_best_practices.md (hunk 6 above)
Comment (PastaHash) [reply docs/guides/validators/security_best_practices.md]: fixed

**Review by kioqq — APPROVED** [review]

File: docs/guides/validators/security_best_practices.md [hunk 7]
```diff
@@ -330,27 +330,58 @@ Once the installation is complete, you can save your configuration using the com
 sudo invoke-rc.d iptables-persistent save
 ```
 
-### Rule for ssh connection
+### One command firewall configuration
```
Comment (onsails) [inline docs/guides/validators/security_best_practices.md]: that's definitely not one command but whole four :)

Comment (onsails) [reply docs/guides/validators/security_best_practices.md]: we should make it clear that this configuration is for validator, not RPC.

**Review by onsails — CHANGES_REQUESTED** [review]

File: docs/guides/validators/security_best_practices.md (hunk 7 above)
Comment (PastaHash) [reply docs/guides/validators/security_best_practices.md]: fixed

**Comment by onsails** [comment]
image
Rebase instead of merging.

**Comment by onsails** [comment]
Now it's 5 commits, 4 with the same message, one is a merge
image

**Review by onsails — APPROVED** [review]
